lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -83,33 +81,34 @@ export abstract class ClientApplication {
          * This is to support server-side rendering environments.
          */
         this.isBrowserEnvironment = typeof window !== "undefined";
-        if (!this.isBrowserEnvironment) {
-            return;
-        }
-
         // Set the configuration.
-        this.config = buildConfiguration(configuration);
+        this.config = buildConfiguration(configuration, this.isBrowserEnvironment);
 
-        // Initialize the crypto class.
-        this.browserCrypto = new CryptoOps();
+        this.defaultAuthority = null;
+        this.activeLocalAccountId = null;
 
-        // Initialize the network module class.
-        this.networkClient = this.config.system.networkClient;
+        // Array of events
+        this.eventCallbacks = new Map();
 
         // Initialize logger
         this.logger = new Logger(this.config.system.loggerOptions, name, version);
 
+        // Initialize the network module class.
+        this.networkClient = this.config.system.networkClient;
+
+        if (!this.isBrowserEnvironment) {
```

Comment:
The changes in the constructor have implications on SSR and I think this creates risk that we may break SSR in the future. Open to feedback on a better way to both ensure SSR is supported and properties on PCA are "definitely" initialized

Reply:
There should be unit tests that make sure the constructor doesnt throw when `window` isn't defined.